### PR TITLE
tests: posix: Fix array overrun issue

### LINKED ISSUE #12309: [Coverity CID :190624]Memory - corruptions in /tests/posix/fs/src/test_fs_file.c
Static code scan issues seen in File: /tests/posix/fs/src/test_fs_file.c
 Category: Memory - corruptions
 Function: test_file_read
 Component: Tests
 CID: 190624
 Please fix or provide comments to square it off in coverity in the link: https://scan9.coverity.com/reports.htm#v32951/p12996

### PR DESCRIPTION
**Pull request by ramakrishnapallala**
Fix Coverity Array overrun issue.

Fixes #12309 

Signed-off-by: Ramakrishna Pallala <ramakrishna.pallala@intel.com>

### CODE REVIEW

**Review by nniranjhana — APPROVED** [review]

File: tests/posix/fs/src/test_fs_file.c [hunk 1]
```diff
@@ -114,6 +114,9 @@ static int test_file_read(void)
 		return TC_FAIL;
 	}
 
+	/* Check for array overrrun */
```
Comment (nniranjhana) [inline tests/posix/fs/src/test_fs_file.c]:
```suggestion
	/* Check for array overrun */
```

**Comment by nashif** [comment]
> Fix Coverity Array overrun issue.

fix the commit message, you are not fixing a "Coverity Array", coverity should not be in the commit title message at all

**Review by nashif — CHANGES_REQUESTED** [review]
fix commit message

**Comment by ramakrishnapallala** [comment]
> fix commit message

Done.

**Comment by aescolar** [comment]
> Done.

@ramakrishnapallala : Not really ;) . @nashif point is very valid, could you address it?

**Comment by ramakrishnapallala** [comment]
> > Done.
> 
> @ramakrishnapallala : Not really ;) . @nashif point is very valid, could you address it?

Removed Coverity from Commit message

**Review by pfalcon — APPROVED** [review]
It would have been better to pass `sizeof(read_buff) - 1` to read(), but given this is a test, well...

**Comment by pfalcon** [comment]
@nashif: Please re-review.

**Review by nashif — APPROVED** [review]

**Pull request merged**